newIDE/app/src/ResourcesList/ResourceSource.flow.js

```diff
@@ -1,13 +1,16 @@
 // @flow
 import * as React from 'react';
 import { type I18n as I18nType } from '@lingui/core';
+import type { PreferencesValues } from '../MainFrame/Preferences/PreferencesContext';
 
 // These are all the kind of resources that can be found in
 // Core/GDCore/Project/ResourcesManager.h
 export type ResourceKind = 'image' | 'audio' | 'font' | 'video' | 'json';
 
 export type ResourceSourceComponentProps = {|
   i18n: I18nType,
+  loadPreferencesValues: () => ?PreferencesValues,
+  savePreferencesValues: (values: PreferencesValues) => void,
```

Comment:
Instead of passing the whole functions that are used to save and load preferences, could you send the preference values and a function to update the path?

=> the idea is to give to a component *nothing more* than what is needed for it to work. The components don't need to be able to change the entire preferences. It's even dangerous - they could erase everything :) Instead, let's see if we can just pass them a function to update the latest path. 

See for example how it's done for the language:
https://github.com/4ian/GDevelop/blob/a315eabdcedb5f04b7bf287272fa46e7b43f924f/newIDE/app/src/MainFrame/Preferences/LanguageDialog.js#L74-L75

we're getting the values (no problem if you want to read them, they are not secret) and *only* a function to set the language: `setLanguage`

Let's do the same here :)